draft-ietf-scitt-architecture.md

```diff
@@ -1095,3 +1106,194 @@ To indicate that the content is an scitt configuration represented as JSON:
+
+SCITT Identifiers for URLs and URNs that are not Data URLs MUST be represented in binary using {{-CURIs}}.
+
+For each SCITT conceptual message, we define a Data URL format according to {{-DataURLs}}, a URN format according to {{-URNs}} and a URL format according to {{URLs}}.
+
+Note that Data URLs require base64 encoding, but the URN definitions require base64url encoding.
+
+Resolution and dereferencing of these identifiers is out of scope for this document, and can be implemented by any concrete api implementing the abstract interface defined as follows:
+
+~~~
+resource: content-type = dereference(identifier: identifier-type)
+~~~
+
+These identifiers MAY be present in a `tstr` field that does not otherwise restrict the string in ways that prevent a URN or URL from being present.
+
+This includes `iss`, and `sub` which are used to express the issuer and subject of a signed statement or receipt.
+
+This also includes `kid` which is used to express a hint for which public key should be used to verify a signature.
+
+All SCITT identifiers share common parameters to promote interoperability:
+
+Let hash-name be an algorithm name registered in {{IANA.named-information}}.
+
+To promote interoperability, the hash-name MUST be "sha-256".
+
+Let base-encoding, be a base encoding defined in {{-Base64Url}}.
+
+To promote interoperability, the base encoding MUST be "base64url".
+
+In the blocks and examples that follow, NOTE: '\' line wrapping per RFC 8792.
+
+## For Binary Content
+
+Identifiers for binary content, such as Statements, or even Artifacts themselves are computed as follows:
+
+Let the `base64url-encoded-bytes-digest` for the messsage be the base64url encoded digest with the chosen hash algorithm of bytes / octets.
+
+Let the SCITT name for the message be the URN constructed from the following URI templatem, according to {{-URITemplate}}:
+
+Let the `message-type`, be "statement" for Statements and Artifacts.
+
+~~~
+urn:ietf:params:scitt:\
+{message-type}:\
+{hash-name}:{base-encoding}:\
+{base64url-encoded-bytes-digest}
+~~~
+
+## For SCITT Messages
+
+Identifiers for COSE Sign 1 based messages, such as identifiers for Signed Statements and Receipts are computed as follows:
+
+Let the `base64url-encoded-to-be-signed-bytes-digest` for the messsage be the base64url encoded digest with the chosen hash algorithm of the "to-be-signed bytes", according to {{Section 8.1 of RFC9052}}.
+
+Let the SCITT name for the message be the URN constructed from the following URI templatem, according to {{-URITemplate}}:
+
+Let the `message-type`, be "signed-statement" for Signed Statements, and "receipt" for Receipts.
+
+~~~
+urn:ietf:params:scitt:\
+{message-type}:\
+{hash-name}:{base-encoding}:\
+{base64url-encoded-to-be-signed-bytes-digest}
+~~~
+
+Note that this means the content of the signature is not included in the identifer, even though signature related claims, such as activation or expiration information in protected headers are included.
```

Comment:
@fournet @ad-l per our call today, please make sure these are the security properties you want for "identifiers" for cose sign 1 messages.